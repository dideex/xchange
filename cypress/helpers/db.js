const mongoose = require('mongoose')
const mongoConfig = require('./config.json').mongodb

module.exports = mongoose.connect(mongoConfig, {useNewUrlParser: true})
