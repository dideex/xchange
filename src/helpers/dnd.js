import React from 'react'
import {DragDropContext} from 'react-dnd'
import TestBackend from 'react-dnd-test-backend'

export function wrapInTestContext(DecoratedComponent) {
  const TestStub = props => <DecoratedComponent {...props} />
  return DragDropContext(TestBackend)(TestStub)
}
