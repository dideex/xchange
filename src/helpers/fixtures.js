export const fakeCurrnecy = [
  {
    order: 1,
    _id: 1,
    icon: 'eth',
    id: 'eth',
    name: 'tether',
    label: 'tether label',
    source: 'tether source wallet'
  },
  {
    order: 2,
    _id: 2,
    icon: 'bitcoin',
    id: 'bitcoin',
    name: 'lisk',
    label: 'lisk label',
    source: 'lisk source wallet'
  },
  {
    order: 3,
    _id: 3,
    icon: 'rur',
    id: 'rur',
    name: 'litecoin',
    label: 'litecoin label',
    source: 'litecoin source wallet'
  },
]


export const fakeData = {
  id: 'Test id',
  inputValue: '1000',
  toWallet: 'Test wallet',
  currencyInputLabel: 'Test input label',
  currencyOutputLabel: 'Test output label',
  fromWallet: 'Tests fromWallet',
  outputValue: '2000',
  email: 'Tests email',
  username: 'Tests username',
  paymentStatus: 1,
  loading: false,
  wallets: {
    eth: 'eth wallet',
    bitcoint: 'bitcoint wallet',
    rur: 'rur wallet',
  },
  login: 'Test login',
  updatePaymentStatus() {},
}

export const fakeUser = {
  username: 'Fake username',
  password: 'Fake password',
  email: 'fake@email.com',
  phone: '89145678901',
  fio: 'Fake fio',
  token: 'Fake token',
}

export const fakeBitcoinAddress = '3JQSigWTCHyBLRD979JWgEtWP5YiiFwcQB '
export const fakeCreditAddress = '1234 4321 5678 8765'

export const fakeWallets = {
  eth: 'Fake eth wallet',
  bitcoin: 'Fake bitcoin wallet',
  rur: 'Fake rur wallet'
}