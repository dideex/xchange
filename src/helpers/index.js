export {fakeCurrnecy} from './fixtures'

export const delay = (fn = () => {}, timeout = 0) =>
  new Promise(res =>
    setTimeout(() => {
      fn()
      res()
    }, timeout),
  )
