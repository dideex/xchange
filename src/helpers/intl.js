import React from 'react'
import {mount, shallow} from 'enzyme'
import {IntlProvider, intlShape} from 'react-intl'

import messages from '../locale/en.json'
const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

function nodeWithIntlProp(node) {
  return React.cloneElement(node, {intl})
}

export function mountWithIntl(
  node,
  {context, childContextTypes, ...additionalOptions} = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, {intl}),
    childContextTypes: Object.assign({}, {intl: intlShape}, childContextTypes),
    ...additionalOptions,
  })
}
