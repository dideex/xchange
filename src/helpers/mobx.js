import React from 'react'
import {Provider} from 'mobx-react'
import {lastOperations, cash, user} from '../store'

jest.mock('../components/Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}}),
  errorEmitter: jest.fn(data => fn => data(fn)),
}))

// jest.mock('../components/common', () => ({
//   noty: () => {},
//   Loading: () => <div>Loading</div>,
// }))

export const MobxProvider = ({children, ...stores}) => {
  return (
    <Provider
      lastOperationsStore={lastOperations}
      cashStore={cash}
      userStore={user}
      {...stores}
    >
      {children}
    </Provider>
  )
}
