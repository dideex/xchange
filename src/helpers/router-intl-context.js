import {mount} from 'enzyme'
import {BrowserRouter} from 'react-router-dom'
import {IntlProvider, intlShape} from 'react-intl'
import {shape} from 'prop-types'

import messages from '../locale/en.json'

export const testId = 'test id'

// Instantiate router context
const router = {
  history: {...new BrowserRouter().history},
  route: {
    location: {
      path: '/location',
      href: '/location',
    },
    match: {
      path: '/location',
      params: {
        id: testId,
      },
    },
  },
}

// const createContext = () => ({
//   context: {router},
//   childContextTypes: {router: shape({})},
// })

const intlProvider = new IntlProvider({locale: 'en', messages}, {})
const {intl} = intlProvider.getChildContext()

const createMixedContext = ({
  router,
  context,
  childContextTypes,
  ...additionalOptions
} = {}) => ({
  context: Object.assign({}, {router}, context, {intl}),
  childContextTypes: Object.assign(
    {},
    {router: shape({})},
    {intl: intlShape},
    childContextTypes,
  ),
  ...additionalOptions,
})

export function mountWrap(node) {
  return mount(node, createMixedContext({router}))
}

// export function shallowWrap(node) {
//   return shallow(node, createContext())
// }
