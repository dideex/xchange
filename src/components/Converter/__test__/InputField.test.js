import React from 'react'
import Component from '../InputField'
import {mount} from 'enzyme'
import {wrapInTestContext} from '../../../helpers/dnd'
import {fakeCurrnecy, fakeUser} from '../../../helpers/fixtures'
import CashStore from '../../../store/Cash'

jest.mock('../../Api', () => ({
  post: () => Promise.resolve({data: {}}),
  get: () => Promise.resolve({data: {}, eth: {}}),
  errorEmitter: fn => data => fn(data),
}))

jest.mock('../../common/Noty.js', () => ({
  noty: () => {},
}))

describe('Coverter: Currency badge tests', () => {
  let cashStore
  beforeEach(() => {
    cashStore = new CashStore()
  })
  const props = {
    inputValue: '10',
    currencyId: 0,
    changeInput: () => {},
    onSelectChange: () => {},
  }
  describe('Markup', () => {
    it('Basic markup', () => {
      const WrappedComponent = wrapInTestContext(Component)
      const wrapper = mount(<WrappedComponent {...props} cashStore={cashStore} />)

      expect(wrapper.html()).toMatchSnapshot()
    })

    it('Currency icon should be', () => {
      const WrappedComponent = wrapInTestContext(Component)
      cashStore.currency = fakeCurrnecy
      const wrapper = mount(<WrappedComponent {...props} cashStore={cashStore} />)

      expect(wrapper.html()).toMatchSnapshot()
    })

    describe('Dropdown behaviour', () => {
      it('Dropdown should open', () => {
        const WrappedComponent = wrapInTestContext(Component)
        cashStore.currency = fakeCurrnecy
        const wrapper = mount(<WrappedComponent {...props} cashStore={cashStore} />)
        wrapper.find('CurrencySelect').simulate('mouseenter')

        expect(wrapper.find('InputField').instance().state.open).toBeTruthy()
        expect(wrapper.html()).toMatchSnapshot()
      })

      it('Dropdown should close', () => {
        const WrappedComponent = wrapInTestContext(Component)
        cashStore.currency = fakeCurrnecy
        const wrapper = mount(<WrappedComponent {...props} cashStore={cashStore} />)
        wrapper.find('CurrencySelect').simulate('mouseenter')

        wrapper
          .find('InputField > div')
          .first()
          .simulate('mouseleave')

        expect(wrapper.find('InputField').instance().state.open).toBeFalsy()
        expect(wrapper.html()).toMatchSnapshot()
      })
    })
  })

  describe('Handlers should invoke', () => {
    it('Change input handler should ivoke', () => {
      const changeInput = jest.fn()
      const WrappedComponent = wrapInTestContext(Component)
      cashStore.currency = fakeCurrnecy
      const wrapper = mount(
        <WrappedComponent {...props} cashStore={cashStore} changeInput={changeInput} />,
      )

      wrapper.find('input').simulate('change', {target: {value: fakeUser.username}})

      expect(changeInput).toHaveBeenCalledTimes(1)
      expect(changeInput).toHaveBeenCalledWith(fakeUser.username)
    })

    it('Select change handler should ivoke', () => {
      const onSelectChange = jest.fn()
      const WrappedComponent = wrapInTestContext(Component)
      cashStore.currency = fakeCurrnecy
      const wrapper = mount(
        <WrappedComponent
          {...props}
          cashStore={cashStore}
          onSelectChange={onSelectChange}
        />,
      )

      wrapper.find('CurrencySelect').simulate('mouseenter')
      wrapper
        .find('CurrencySelect .dropdown__content Label')
        .first()
        .simulate('click')

      expect(onSelectChange).toHaveBeenCalledTimes(1)
      expect(onSelectChange).toHaveBeenCalledWith(fakeCurrnecy[0].id)
    })
  })

  describe('Drag and drop behaviour', () => {
    // TODO: add react dnd implementation
    // https://github.com/react-dnd/react-dnd/blob/fd1bae42a9/packages/dnd-core/src/__tests__/DragDropMonitor.spec.ts
    it.skip('Inputs should change styles when dragging', () => {
      const WrappedComponent = wrapInTestContext(Component)
      cashStore.currency = fakeCurrnecy
      const wrapper = mount(<WrappedComponent {...props} cashStore={cashStore} />)
      const backend = wrapper
        .instance()
        .getManager()
        .getBackend()
      backend.simulateBeginDrag([null])
      backend.simulateEndDrag()
      console.log('TCL: backend', backend)
      // console.log(wrapper.debug())
      // backend.simulateBeginDrag([
      //   wrapper
      //     .find('DragSource(CurrencyBadge)')
      //     .instance()
      //     .getHandlerId(),
      // ])
    })
  })
})

/* 

const WrappedComponent = wrapInTestContext(Component)

      const wrapper = mount(<WrappedComponent name="Ethereum" id={123} icon="Ethereum" />)

      const backend = wrapper
        .instance()
        .getManager()
        .getBackend()
  
      const backend = wrapper
        .instance()
        .getManager()
        .getBackend()

  // Test that the opacity is 1
  let div = TestUtils.findRenderedDOMComponentWithTag(root, 'div')
  expect(div.props.style.opacity).toEqual(1)

  // Find the drag source ID and use it to simulate the dragging operation
  const box = TestUtils.findRenderedComponentWithType(root, Box)
  backend.simulateBeginDrag([box.getHandlerId()])

  // Verify that the div changed its opacity
  div = TestUtils.findRenderedDOMComponentWithTag(root, 'div')
  expect(div.style.opacity).toEqual(0.4)
  */
