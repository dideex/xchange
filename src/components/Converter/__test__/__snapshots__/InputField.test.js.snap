// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coverter: Currency badge tests Markup Basic markup 1`] = `
"<div><div style=\\"background-color: rgb(174, 226, 57);\\" class=\\"css-26z1jj\\"><span class=\\"css-o79fz0\\"></span><label class=\\"css-gudk1o\\"><input style=\\"padding: 10px 34px; text-align: center; width: 100%;\\" placeholder=\\"value\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"10\\"></label><div class=\\"dropdown css-1ft27gx\\"><svg style=\\"width: 32px; height: 32px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1
 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg></div><span class=\\"css-1sr8xmg\\">0</span></div></div>"
`;

exports[`Coverter: Currency badge tests Markup Currency icon should be 1`] = `
"<div><div style=\\"background-color: rgb(174, 226, 57);\\" class=\\"css-26z1jj\\"><span class=\\"css-o79fz0\\"><svg class=\\"\\" style=\\"width: 32px; height: 32px;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>bitcoin</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>bitcoin</text></svg></span><label class=\\"css-gudk1o\\"><input style=\\"padding: 10px 34px; text-align: center; width: 100%;\\" placeholder=\\"value\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"10\\"></label><div class=\\"dropdown css-1ft27gx\\"><svg style=\\"width: 32px; height: 32px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1
 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg></div><span class=\\"css-1sr8xmg\\"></span></div></div>"
`;

exports[`Coverter: Currency badge tests Markup Dropdown behaviour Dropdown should close 1`] = `
"<div><div style=\\"background-color: rgb(174, 226, 57);\\" class=\\"css-26z1jj\\"><span class=\\"css-o79fz0\\"><svg class=\\"\\" style=\\"width: 32px; height: 32px;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>bitcoin</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>bitcoin</text></svg></span><label class=\\"css-gudk1o\\"><input style=\\"padding: 10px 34px; text-align: center; width: 100%;\\" placeholder=\\"value\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"10\\"></label><div class=\\"dropdown css-1ft27gx\\"><svg style=\\"width: 32px; height: 32px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1
 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg><div class=\\"css-tmmt7l content--exit content--exit-active\\"><span style=\\"color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>bitcoin</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>bitcoin</text></svg><span style=\\"padding-left: 45px;\\">lisk</span></span><span style=\\"color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>eth</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>eth</text></svg><span style=\\"padding-left: 45px;\\">tether</span></span><span style=\\"color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>rur</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>rur</text></svg><span style=\\"padding-left: 45px;\\">litecoin</span></span></div></div><span class=\\"css-1sr8xmg\\"></span></div></div>"
`;

exports[`Coverter: Currency badge tests Markup Dropdown behaviour Dropdown should open 1`] = `
"<div><div style=\\"background-color: rgb(174, 226, 57);\\" class=\\"css-26z1jj\\"><span class=\\"css-o79fz0\\"><svg class=\\"\\" style=\\"width: 32px; height: 32px;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>bitcoin</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>bitcoin</text></svg></span><label class=\\"css-gudk1o\\"><input style=\\"padding: 10px 34px; text-align: center; width: 100%;\\" placeholder=\\"value\\" type=\\"text\\" mask=\\"\\" class=\\"css-18b8ap6\\" value=\\"10\\"></label><div class=\\"dropdown css-1ft27gx\\"><svg style=\\"width: 32px; height: 32px;\\" class=\\"\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><path style=\\"fill: #aee239;\\" d=\\"M184.9,344.1L8.2,167.4c-8.5-8.5-8.5-22.3,0-30.9l20.6-20.6c8.5-8.5,22.3-8.5,30.8,0l140.7,140.1l140.7-140.1
 c8.5-8.5,22.3-8.5,30.8,0l20.6,20.6c8.5,8.5,8.5,22.3,0,30.9L215.8,344.1C207.3,352.6,193.5,352.6,184.9,344.1L184.9,344.1z\\"></path></svg><div class=\\"dropdown__content css-tmmt7l content--enter content--enter-active\\"><span style=\\"color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>bitcoin</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>bitcoin</text></svg><span style=\\"padding-left: 45px;\\">lisk</span></span><span style=\\"color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>eth</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>eth</text></svg><span style=\\"padding-left: 45px;\\">tether</span></span><span style=\\"color: rgb(25, 23, 22); background: transparent;\\" class=\\"css-t2xfrr\\"><svg class=\\"currency_coin\\" style=\\"width: 32px; height: 32px; z-index: 1; opacity: 1;\\" version=\\"1.1\\" x=\\"0px\\" y=\\"0px\\" viewBox=\\"0 0 400 400\\"><title>rur</title><circle style=\\"fill: #edda48;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"199.5\\"></circle><circle style=\\"fill: #e2b255;\\" cx=\\"199.5\\" cy=\\"200.5\\" r=\\"168.3\\"></circle><text>rur</text></svg><span style=\\"padding-left: 45px;\\">litecoin</span></span></div></div><span class=\\"css-1sr8xmg\\"></span></div></div>"
`;
